cloudsql/appengine/flexible/Startup.cs

```diff
@@ -0,0 +1,222 @@
+                    options.ProjectId = projectId;
+                    options.ServiceName = "CloudSqlSample";
+                    options.Version = "Test";
+                });
+            }
+            services.AddSingleton(typeof(DbConnection), (IServiceProvider) =>
+                InitializeDatabase());
+            services.AddMvc(options =>
+            {
+                options.Filters.Add(typeof(DbExceptionFilterAttribute));
+            });
+            _services = services;
+        }
+
+        DbConnection InitializeDatabase()
+        {
+            DbConnection connection;
+            string database = Configuration["CloudSQL:Database"];
+            switch (database.ToLower())
+            {
+                case "mysql":
+                    connection = GetMySqlConnection();
+                    break;
+                case "postgresql":
+                    connection = GetPostgreSqlConnection();
+                    break;
+                default:
+                    throw new ArgumentException(string.Format(
+                        "Invalid database {0}.  Fix appsettings.json.",
+                            database), "CloudSQL:Database");
+            }
+            connection.Open();
+            using (var createTableCommand = connection.CreateCommand())
+            {
+                createTableCommand.CommandText = @"
+                    CREATE TABLE IF NOT EXISTS
+                    votes(
+                        vote_id SERIAL NOT NULL,
+                        time_cast timestamp NOT NULL,
+                        candidate CHAR(6) NOT NULL,
+                        PRIMARY KEY (vote_id)   
+                    )";
+                createTableCommand.ExecuteNonQuery();
+            }
+            return connection;
+        }
+
+        DbConnection NewMysqlConnection()
+        {
+            // [START cloud_sql_mysql_connection_pool]
+            var connectionString = new MySqlConnectionStringBuilder(
+                Configuration["CloudSql:ConnectionString"])
+            // ConnectionString set in appsetings.json formatted as:
+            // "Uid=aspnetuser;Pwd=;Host=cloudsql;Database=votes"
+            {
+                // Connecting to a local proxy that does not support ssl.
+                SslMode = MySqlSslMode.None,
+            };
+            connectionString.Pooling = true;
+            // [START_EXCLUDE]
+            // [START cloud_sql_mysql_max_connections]
+            connectionString.MaximumPoolSize = 5;
+            // [END cloud_sql_mysql_max_connections]
+            // [START cloud_sql_mysql_connection_timeout]
+            connectionString.ConnectionTimeout = 15;
+            // [END cloud_sql_mysql_connection_timeout]
+            // [START cloud_sql_mysql_connection_lifetime]
+            connectionString.ConnectionLifeTime = 300;
+            // [END cloud_sql_mysql_connection_lifetime]
+            // [END_EXCLUDE]
+            DbConnection connection =

```

Comment:
Is this a pool, or is it a connection? 

Can we add either some additional comments or perhaps a link explaining how connections are reused, and what the best practices around closing or retrieving is?